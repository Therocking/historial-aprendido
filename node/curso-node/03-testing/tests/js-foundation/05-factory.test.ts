import {buildMakePerson} from '../../src/js-foundation/05-factory';

describe('05-factory', () => {

    const getUUID = () => '123456'
    const getAge = () => 18
    
    test('makePerson should macth to snapshot', () => {
        const makePerson = buildMakePerson({ getUUID, getAge });

        expect(makePerson).toMatchSnapshot()

    })

    test('buildMakePerson should be a fn', () => {

        const makePerson = buildMakePerson({ getUUID, getAge });

        expect( typeof makePerson ).toBe('function')
    })

    test('should returns a person obj', () => {

        const obj = {
            name: 'jose',
            birthdate: '2005-7-24'
        }

        const makePerson = buildMakePerson({ getUUID, getAge });
        const person = makePerson(obj);

        expect( person ).toEqual({
            id: '123456',
            name: 'jose',
            birthdate: '2005-7-24',
            age: 18
        });
    });
});