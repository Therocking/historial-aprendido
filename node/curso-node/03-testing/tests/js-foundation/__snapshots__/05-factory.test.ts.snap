// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`05-factory makePerson should macth to snapshot 1`] = `[Function]`;
